mkt/tags/models.py

```diff
@@ -57,3 +56,23 @@ class AddonTag(ModelBase):
 
     class Meta:
         db_table = 'users_tags_addons'
+
+
+def attach_tags(objs, m2m_name=None):
+    """
+    Fetch tags from `objs` in one query and then attach them to a property on
+    each instance. The name of the property will be `m2m_name` + '_list'.
+
+    Assumes every instance in `objs` uses the same model, and needs `m2m_name`
+    to be set to the name of the m2m field between the model and Tag.
+    """
+    if m2m_name is None:
+        raise ValueError('m2m_name needs to be set.')
+    if objs:
+        obj_dict = {obj.id: obj for obj in objs}
+        field_name = getattr(objs[0], m2m_name).query_field_name
+        qs = (Tag.objects.not_blocked()
+              .filter(**{'%s__in' % field_name: obj_dict})
```

Comment:
This string formatting is a bit hairy but I'm not sure I would change it. Would be be better to pass `obj_dict.keys()` here? I'm assuming it ends up getting iterated over and it works as a `dict` but it seems clearer.